tests/test_widget.py

```diff
@@ -133,3 +189,43 @@ def test_calculate_length_fails_with_wrong_number_of_points(
         "ERROR: Select two points to calculate the decay length."
         in captured.out
     )
+
+
+@pytest.mark.parametrize(
+    "image_folder",
+    [
+        "./test_data_view1",
+    ],
+)
+def test_load_data(cpt_widget, qtbot, image_folder):
+    """Test loading of images in a folder as stack associated to a certain view"""
+
+    os.mkdir(image_folder)
+    data = np.random.randint(0, 255, (256, 256, 3), "uint8")
+    tf.imwrite(image_folder + "/temp1.tif", data, photometric="rgb")
+    data = np.random.randint(0, 255, (256, 256, 3), "uint8")
+    tf.imwrite(image_folder + "/temp2.tif", data, photometric="rgb")
+
+    def set_directory_and_close(dialog):
+        qtbot.addWidget(dialog)
+        dialog.setDirectory(image_folder)
+        buttonbox = dialog.findChild(QDialogButtonBox, "buttonBox")
+        openbutton = buttonbox.children()[1]
+        qtbot.mouseClick(openbutton, Qt.LeftButton, delay=1)
+
+    # Open and retrieve file dialog
+    get_dialog(
+        cpt_widget.load.setCurrentIndex,
+        set_directory_and_close,
+        time_out=5,
+        trigger_option=1,
```

Comment:
Cool 😎  , but I'd explicitly give the keyword arguments for clarity...

```suggestion
        dialog_trigger=cpt_widget.load.setCurrentIndex,
        dialog_action=set_directory_and_close,
        time_out=5,
        trigger_option=1,
```